packages/api-derive/src/tx/signingInfo.ts

```diff
@@ -84,8 +84,8 @@ export function signingInfo (_instanceId: string, api: DeriveApi): (address: str
           api.consts.system?.blockHashCount?.toNumber() || FALLBACK_MAX_HASH_COUNT,
           MORTAL_PERIOD
             .div(
-              api.consts.babe?.expectedBlockTime ||
-              api.consts.timestamp?.minimumPeriod.muln(2) ||
+              (!api.consts.babe?.expectedBlockTime.isZero() && api.consts.babe?.expectedBlockTime) ||
+              (!api.consts.aura?.slotDuration.isZero() && api.consts.aura.slotDuration) ||
```

Comment:
Maybe is better to use a helper function to check that is not `zero` for both (babe/aura) fields.

Reply:
Yea this makes sense! WIll help clean it up a bit